### PR TITLE
Add the bd_s390_dasd_is_fba function to check if DASD is FBA

### PR DESCRIPTION
**Pull request by poncovka**
This will enable blivet to choose a correct disklabel type for an FBA device.

### CODE REVIEW

File: src/plugins/s390.c [hunk 1]
```diff
@@ -290,6 +290,52 @@ gboolean bd_s390_dasd_is_ldl (const gchar *dasd, GError **error) {
...
+ * Returns: whether a dasd is FBA
+ */
+gboolean bd_s390_dasd_is_fba (const gchar *dasd, GError **error) {
+    gchar *devname = NULL;
+    gint f = 0;
+    gint blksize = 0;
+    dasd_information2_t dasd_info;
+
+    memset(&dasd_info, 0, sizeof(dasd_info));
+    devname = g_strdup_printf ("/dev/%s", dasd);
```
Comment (vpodzime) [inline src/plugins/s390.c]: Please check that ``dasd`` doesn't already start with ``"/dev/"`` here.

File: src/plugins/s390.c [hunk 2]
```diff
@@ -290,6 +290,52 @@ gboolean bd_s390_dasd_is_ldl (const gchar *dasd, GError **error) {
...
+        g_free (devname);
+        return FALSE;
+    }
+
+    g_free (devname);
+
+    if (ioctl(f, BLKSSZGET, &blksize) != 0) {
+        close(f);
+        return FALSE;
+    }
```
Comment (vpodzime) [inline src/plugins/s390.c]: What is the ``blksize`` good for? I cannot see it being used anywhere below.

Comment (poncovka) [inline src/plugins/s390.c]: The function `bd_s390_dasd_is_fba` is copy-pasted from  `bd_s390_dasd_is_ldl`. The corresponding code for RHEL 7 is here: https://github.com/rhinstaller/blivet/blob/rhel7-branch/blivet/devicelibs/dasd.py#L126. There is a comment "poorly check if this is even a block device...".

File: src/plugins/s390.c (hunk 1 above)
Comment (poncovka) [inline src/plugins/s390.c]: Should I also change `bd_s390_dasd_is_ldl`? (see the comment below)

Comment (vpodzime) [inline src/plugins/s390.c]: Yes please.

File: src/plugins/s390.c (hunk 2 above)
Comment (vpodzime) [inline src/plugins/s390.c]: Oh, then this should have the same comment justifying the reason for doing the ``ioctl()``. (and the same applies to ``is_ldl()``). Thanks!

File: src/plugins/s390.c (hunk 1 above)
Comment (poncovka) [inline src/plugins/s390.c]: Fixed.

File: src/plugins/s390.c (hunk 2 above)
Comment (poncovka) [inline src/plugins/s390.c]: Fixed.

File: src/plugins/s390.c [hunk 3]
```diff
@@ -290,6 +290,62 @@ gboolean bd_s390_dasd_is_ldl (const gchar *dasd, GError **error) {
...
+
+    close(f);
+
+    /* check if we're on an FBA DASD */
+    if (strncmp(dasd_info.type, "FBA", 3) == 0) {
+        return TRUE;
+    }
+    else {
+        return FALSE;
```
Comment (vpodzime) [inline src/plugins/s390.c]: Please replace this ``if`` with ``return strncmp(dasd_info.type, "FBA", 3) == 0``.

Comment (poncovka) [inline src/plugins/s390.c]: Fixed.

**Pull request merged**